app/policies/request_policy.rb

```diff
@@ -4,26 +4,24 @@ class Scope < ApplicationPolicy::Scope
     PERSONA_APPROVER  = 'approval/approver'.freeze
     PERSONA_REQUESTER = 'approval/requester'.freeze
 
-    def resolve
-      return scope.all unless user.rbac_enabled?
-
+    def resolve_scope
       if user.params[:request_id]
         req = Request.find(user.params[:request_id])
         raise Exceptions::NotAuthorizedError, "Read access not authorized for request #{req.id}" unless resource_check('read', req)
-        req.children
-      # GraphQL comes in with model_class.all, whose scope is respond to model call
-      # Regular root requests call comes in with model class itself
+
+        req.requests
       else
         case Insights::API::Common::Request.current.headers[Insights::API::Common::Request::PERSONA_KEY]
         when PERSONA_ADMIN
           raise Exceptions::NotAuthorizedError, "No permission to access the complete list of requests" unless admin?(scope)
-          scope.respond_to?(:model) ? scope : scope.where(:parent_id => nil)
+
+          scope.root_requests_only
         when PERSONA_APPROVER
           raise Exceptions::NotAuthorizedError, "No permission to access requests assigned to approvers" unless approver?(scope)
-          scope.respond_to?(:model) ? (raise Exceptions::NotAuthorizedError, "Approver not allowed to access requests via GraphQL") :
-                                      approver_visible_requests(scope)
+
+          approver_visible_requests(scope).child_requests_only
```

Comment:
should not and cannot have `.child_requests_only` because qualified requests may include root requests if they don't have children.

Reply:
👍